### PR TITLE
Implement native methods for RuntimeEffect

### PR DESCRIPTION
**Pull request by VeselovAlex**

### CODE REVIEW

File: skiko/src/nativeJsMain/cpp/RuntimeEffect.cc [hunk 1]
```diff
@@ -5,75 +5,62 @@
...
     }
-    env->ReleaseLongArrayElements(childrenPtrsArr, childrenPtrs, 0);
 
     sk_sp<SkShader> shader = runtimeEffect->makeShader(sk_ref_sp<SkData>(uniform),
                                                        children.data(),
                                                        childCount,
                                                        localMatrix.get(),
                                                        isOpaque);
-    return ptrToKNativePointer(shader.release());
+    return reinterpret_cast<KNativePointer>(shader.release());
```
Comment (olonho) [inline skiko/src/nativeJsMain/cpp/RuntimeEffect.cc:25]: Technically this cast is not required.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt [hunk 2]
```diff
@@ -109,8 +109,8 @@ inline fun withResult(result: NativePointerArray, block: (InteropPointer) -> Uni
 }
 
 @Suppress("NON_PUBLIC_CALL_FROM_PUBLIC_INLINE")
-inline fun withStringResult(block: () -> NativePointer): String {
-    val string = ManagedString(block())
+inline fun withStringResult(managed: Boolean = true, block: () -> NativePointer): String {
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:112]: Why would we want `managed = false` here?

Comment (VeselovAlex) [reply skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:112]: Some SkString instances are not owned by another objects, we are receiving them by const&. So it is UB to delete them in ManagedString

Comment (olonho) [reply skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:112]: For such cases we shall hold reference to those objects, as in here https://github.com/JetBrains/skiko/blob/09aad6044e04a0b87ae4a9489de9c2236417d735/skiko/src/commonMain/kotlin/org/jetbrains/skia/Pixmap.kt#L12

Comment (VeselovAlex) [reply skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt:112]: This requires a special Managed class just to keep temporary objects. IMHO it is an overkill

**Review by olonho — APPROVED** [review]

**Pull request merged**